docs/quick-starts/rug-typescript.md

```diff
@@ -0,0 +1,203 @@
+## Rug TypeScript Quick Start
+
+Rugs can be written in [TypeScript](https://www.typescriptlang.org/) or the [Rug DSL](../refence-docs/rug/index.md). The TypeScript approach is usually preferable over the [Rug DSL](../refence-docs/rug/index.md) when you have some real programmatic work to do in your Rugs where the DSL is too constraining. Also using TypeScript has the significant added bonuses of being type-safe and having great existing tooling support for features like code-completion.
+
+In this Quick Start you're going to set up your Rug archive for writing and testing Rugs written in TypeScript and using the [Rug CLI](rug-cli.md).
+
+### Getting set up for TypeScript Rugs
+
+To enable your Rug archive for TypeScript all you need to do is:
+
+-   Add a [standard `package.json` file](https://docs.npmjs.com/files/package.json) into the `.atomist` directory, amending for your own Rug archive's project settings
+-   Add a [standard `tsconfig.json` file](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) into your `.atomist` directory
+-   Install `node` and `npm` for local TypeScript development and, specifically, to install any dependencies your Rug TypeScript sources may have.
+
+You can have both Rug DSL and Rug TypeScript files in the *same* Rug archive. A good example of this is available in the [`travis-editors` Rug archive](https://github.com/atomist-rugs/travis-editors).
+
+> ***NOTE:*** A Rug editor and generator for this work is being produced and will reside in the `rug-editors` Rug archive.
+
+#### Adding a `package.json` file
+
+As a starting point, simply add a [standard `package.json` file](https://docs.npmjs.com/files/package.json) to your `.atomist` directory that contains the following:
+
+```
+{
+  "name": "@atomist-rugs/travis-editors",
+  "version": "0.8.0",
+  "description": "Editors for enabling Travis CI",
+  "keywords": [
+    "Atomist",
+    "Rug",
+    "Travis"
+  ],
+  "author": "Atomist, Inc",
+  "license": "Apache-2.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/atomist-rugs/travis-editors.git"
+  },
+  "bugs": {
+    "url": "https://github.com/atomist-rugs/travis-editors/issues"
+  },
+  "homepage": "https://github.com/atomist-rugs/travis-editors#readme",
+  "dependencies": {
+    "@atomist/travis": "0.5.1",
+    "@atomist/rug": "0.10.0"
+  },
+  "scripts": {
+    "rug-install": "rug install -urX",
+    "rug-test": "rug test -urX",
+    "rug-publish": "rug publish -urX"
+  }
+
+```
+
+Amend the following in this `package.json` file according to your own Rug archive project's settings in your `.atomist/manifest.yml` file.
```

Comment:
Amend what following? Perhaps rather than having `atomist-rugs` and `travis-editors` in the package.json example, put in some placeholders that are easy to spot and replace, e.g., RUG_GROUP and RUG_ARCHIVE.